dnsdist: Fails to build with Boost 1.69 (develop snapshot)

After boostorg/logic@23cd89d4c80f build fails. See [error log](https://ptpb.pw/c1ye).
```c++
dnsdist-lua.cc:76:10: error: no viable conversion from returned value of type 'boost::logic::tribool' to function return type 'bool'
  return g_noLuaSideEffect==true;
         ^~~~~~~~~~~~~~~~~~~~~~~
```

--- a/pdns/dnsdist-lua.cc
+++ b/pdns/dnsdist-lua.cc
@@ -75,7 +75,10 @@ void setLuaSideEffect()
 
 bool getLuaNoSideEffect()
 {
-  return g_noLuaSideEffect==true;
+  if (g_noLuaSideEffect) {
+    return true;
+  }
+  return false;
 }
 
 void resetLuaSideEffect()

